### PR TITLE
Update `LightningQubit` to adhere to MCM qnode arguments

### PR DESCRIPTION
**Pull request by mudit2812**
**Context:**
As name says. `QNode` now accepts `postselect_shots` and `mcm_method` arguments to specify MCM behaviour. This PR updates LQ to adhere to those arguments.

**Description of the Change:**
* Update LQ preprocessing to adhere to user requested MCM behaviour
* Add tests for LQ and LK
* Update LQ and LK simulation to adhere to specified `postselect_mode`.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Comment by mudit2812** [comment]
[sc-63641] [sc-63640]

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 pip~=22.0
-git+https://github.com/PennyLaneAI/pennylane.git@master
+git+https://github.com/PennyLaneAI/pennylane.git@postselect-choice
```
Comment (mudit2812) [inline requirements-dev.txt:2]:
```suggestion
git+https://github.com/PennyLaneAI/pennylane.git@master
```

File: tests/new_api/test_device.py [hunk 2]
```diff
@@ -259,7 +259,11 @@ def test_preprocess(self, adjoint):
         expected_program.add_transform(validate_measurements, name=device.name)
         expected_program.add_transform(validate_observables, accepted_observables, name=device.name)
         expected_program.add_transform(validate_device_wires, device.wires, name=device.name)
-        expected_program.add_transform(mid_circuit_measurements, device=device)
+        expected_program.add_transform(
+            mid_circuit_measurements,
+            device=device,
+            mcm_config={"postselect_shots": None, "mcm_method": None},
```
Comment (trbromley) [inline tests/new_api/test_device.py:265]: Sorry @mudit2812! There is still some discussion on the name for `postselect_shots`, but I hope to resolve that today.

Comment (mudit2812) [reply tests/new_api/test_device.py:265]: Thanks for letting me know!

**Review by vincentmr — COMMENTED** [review]
Nothing to flag here. I think a couple tests needed fixing, but this looks ready to merge. I'll approve once the CI goes green.

**Review by vincentmr — APPROVED** [review]
LGTM, thanks @mudit2812 .

**Review by maliasadi — APPROVED** [review]
Thanks @mudit2812 :raised_hands:

**Pull request merged**